src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -610,10 +758,17 @@ ss::future<response_ptr> describe_configs_handler::handle(
                 ctx.metadata_cache().get_default_retention_local_target_ms()),
               topic_property_retention_local_target_ms,
               topic_config->properties.retention_local_target_ms,
-              request.data.include_synonyms);
+              request.data.include_synonyms,
+              maybe_make_documentation(
+                request.data.include_documentation,
+                config::shard_local_cfg()
+                  .retention_local_target_ms_default.desc()));
 
             if (config_property_requested(
                   resource.configuration_keys, topic_property_remote_delete)) {
+                // Doc string taken from src/v/storage/ntp_config.h
+                docstring = "Controls whether topic deletion should imply "
```

Comment:
Prefer not to have mutable variables lying around and modified if they aren't 100% necessary. These docstrings can be passed as rval parameters to the method:
```
maybe_make_documentation(true_or_false, "this is my docstring");
```